linode_api4/linode_client.py

```diff
@@ -1433,6 +1433,620 @@ def cancel(self):
+        """
+        result = self.client.get("/object-storage/transfer")
+
+        if not "used" in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when getting Transfer Pool!"
+            )
+
+        return MappedObject(**result)
+
+    def buckets(self, *filters):
+        """
+        Returns a paginated list of all Object Storage Buckets that you own.
+        This endpoint is available for convenience.
+        It is recommended that instead you use the more fully-featured S3 API directly.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-buckets-list
+
+        :returns: A list of Object Storage Buckets that matched the query.
+        :rtype: PaginatedList of ObjectStorageBucket
+        """
+        return self.client._get_and_filter(ObjectStorageBucket, *filters)
+
+    def bucket_create(self, cluster, label, acl="private", cors_enabled=False):
+        """
+        Creates an Object Storage Bucket in the specified cluster. Accounts with
+        negative balances cannot access this command. If the bucket already exists
+        and is owned by you, this endpoint returns a 200 response with that bucket
+        as if it had just been created.
+
+        This endpoint is available for convenience.
+        It is recommended that instead you use the more fully-featured S3 API directly.
+
+        API Documentation: https://www.linode.com/docs/api/object-storage/#object-storage-bucket-create
+
+        :param acl: The Access Control Level of the bucket using a canned ACL string.
+                    For more fine-grained control of ACLs, use the S3 API directly.
+        :type acl: str
+                    Enum: private,public-read,authenticated-read,public-read-write
+
+        :param cluster: The ID of the Object Storage Cluster where this bucket
+                        should be created.
+        :type cluster: str
+
+        :param cors_enabled: If true, the bucket will be created with CORS enabled for
+                             all origins. For more fine-grained controls of CORS, use
+                             the S3 API directly.
+        :type cors_enabled: boolean
+
+        :param label: The name for this bucket. Must be unique in the cluster you are
+                      creating the bucket in, or an error will be returned. Labels will
+                      be reserved only for the cluster that active buckets are created
+                      and stored in. If you want to reserve this bucket’s label in
+                      another cluster, you must create a new bucket with the same label
+                      in the new cluster.
+        :type label: str
+
+        :returns: A Object Storage Buckets that created by user.
+        :rtype: ObjectStorageBucket
+        """
+        if acl not in (
+            "private",
+            "public-read",
+            "authenticated-read",
+            "public-read-write",
+        ):
```

Comment:
Umm, in our case I believe that we need to make them a `StrEnum`, but it needs python 3.11 support.

Reply:
Instead, I added helper functions to check if the acl passed in is valid.

Reply:
I think maybe a simple class can do the trick?
```python
class ACL:
    PRIVATE = "private"
    PUBLIC_READ = "public-read"
    AUTHENTICATED_READ = "authenticated-read"
    PUBLIC_READ_WRITE = "public-read-write"
```

Also, there is another ACL, `custom`, https://www.linode.com/docs/api/object-storage/#object-storage-bucket-access-modify__request-body-schema

I think this was added recently by the API team? I can't recall I saw it before...

Reply:
Oh `custom` is added for bucket as well. I believe the updated just happened since I've been keeping watching this doc these days... let me update the ACLs for bucket. And I believe we need to update the ACL in linodego as well. I can create a ticket to follow up.